hack/xref-quadlet-docs

```diff
@@ -0,0 +1,261 @@
+our $Doc = 'docs/source/markdown/podman-systemd.unit.5.md';
+
+# END   user-customizable section
+###############################################################################
+
+###############################################################################
+# BEGIN boilerplate args checking, usage messages
+
+sub usage {
+    print  <<"END_USAGE";
+Usage: $ME [OPTIONS]
+
+$ME cross-checks quadlet documentation between the Go source[Go]
+and the man page[MD].
+
+ [Go]: $Go
+ [MD]: $Doc
+
+We check that:
+
+  * all keys in [Go] are documented in [MD]
+  * all keys in [MD] exist in [Go]
+    * any keys listed in [MD] tables also have a description block
+      and vice-versa
+  * all keys everywhere are in sorted order
+
+OPTIONS:
+
+  --help         display this message
+  --version      display program name and version
+END_USAGE
+
+    exit;
+}
+
+# Command-line options.  Note that this operates directly on @ARGV !
+our $debug   = 0;
+sub handle_opts {
+    use Getopt::Long;
+    GetOptions(
+        'debug!'     => \$debug,
+
+        help         => \&usage,
+        man          => \&man,
+        version      => sub { print "$ME version $VERSION\n"; exit 0 },
+    ) or die "Try `$ME --help' for help\n";
+}
+
+# END   boilerplate args checking, usage messages
+###############################################################################
+
+############################## CODE BEGINS HERE ###############################
+
+# The term is "modulino".
+__PACKAGE__->main()                                     unless caller();
+
+# Main code.
+sub main {
+    # Note that we operate directly on @ARGV, not on function parameters.
+    # This is deliberate: it's because Getopt::Long only operates on @ARGV
+    # and there's no clean way to make it use @_.
+    handle_opts();                      # will set package globals
+
+    # No command-line args
+    die "$ME: Too many arguments; try $ME --help\n"                 if @ARGV;
+
+    my $errs = 0;
+    $SIG{__WARN__} = sub {
+        print STDERR "@_";
+        ++$errs;
+    };
+
+    # Assume that Go source file has Truth
+    my $true_keys = read_go($Go);
+
+    # Read md file, compare against Truth
+    crossref_doc($Doc, $true_keys);
+
+    exit $errs;
+}
+
+
+#############
+#  read_go  #  Returns list of key strings found in quadlet.go
+#############
+sub read_go {
+    my $path = shift;
+    open my $fh, '<', $path
+        or die "$ME: Cannot read $path: $!\n";;
+
+    my @found;                          # List of key strings
+    my $last_constname;                 # Most recently seen const name
+
+    while (my $line = <$fh>) {
+        # Only interested in lines of the form   KeyFoo = "Foo"
+        if ($line =~ /^\s+Key(\S+)\s+=\s+"(\S+)"/) {
+            my ($constname, $keystring) = ($1, $2);
+
+            # const name *should* be the same as the string, except (sigh)
+            # there's an inexplicable block of KeyNetworkFoo = "Foo".
```

Comment:
"inexplicable" to me. If there's a great reason for it, please lmk so I can update the comment.

Reply:
When I added these keys the idea was to indicate that they relate the `.network` files (and hence used in `ConvertNetwork`). I don't see any harm in removing `Network` if you want